feature_engine/selection/shuffle_features.py

```diff
@@ -0,0 +1,213 @@
+        Estimator can be a classifier or regressor
+        
+    scoring: str
+        metric used to evaluate to determine if feature needs to be kept or removed.
+
+    threshold: float
+        the value that defines if a feature will be kept or removed.
+
+    regression: boolean
+        Indicates whether the discretiser should train a regression or a classification
+        decision tree.
+
+    cv: boolean
+         Indicates whether cross-validation will be applied.
+    
+    """
+
+    def __init__(self,
+                 regression,
+                 cv = False,
+                 threshold = 0,
+                 features_to_shuffle = None,
+                 estimator = None,
+                 scoring = None):
+        
+        
+        if not isinstance(cv, bool):
+            raise ValueError("cv can only take True or False")
+
+        if not isinstance(regression, bool):
+            raise ValueError("regression can only take True or False")
+ 
+        self.features_to_shuffle = \
+            _define_variables(features_to_shuffle)
+    
+        self.estimator = estimator
+        
+        self.scoring = scoring
+        
+        self.threshold = threshold
+        
+        self.regression = regression
+        
+        self.cv = cv
+
+    def fit(self, X, y):
+        """
+
+        Parameters
+        ----------
+        
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: array-like of shape (n_samples) 
+            Target variable. Required to train the estimator.
+
+
+        Attributes
+        ----------
+
+        shuffled_features_: dict
+            The shuffled features values
+
+        """
+
+        # check input dataframe
+
+        X = _is_dataframe(X)
+        
+        # find all variables or check those entered are in the dataframe
+        self.features_to_shuffle = _find_all_variables(X, self.features_to_shuffle)
+        
+        # Fit machine learning model with the input estimator if provided.
+        # If the estimator is not provided, default to random tree model 
+        # depending on value of self.regression
+        
+        if (self.estimator is not None):
+            
+            model = self.estimator.fit(X,
+                                       y)
+            y_pred = 1
+            model_performance = 1
+            
+        elif (self.regression):
+            
+            model = RandomForestRegressor().fit(X,
+                                                y)
+            y_pred = 1
+            model_performance = 1
+            
+        else:
+            
+            model = RandomForestClassifier().fit(X,
+                                                 y)
+            y_pred = model.predict_proba(X)[:, 1]
+            model_performance = roc_auc_score(y,
+                                              y_pred)
+
+
+        
+        # dict to collect features and their performance_drift 
+        
```

Comment:
To train the estimator, could we explore the use of cross_validate:
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html

I think if we set return_estimator=True, it returns the trained estimator (1 estimator per cv fold). Then, I understand we could use this to calculate the predictions of the shuffled features

Cross_validate will automatically check that the metric (scoring) and the estimator are valid sklearn functions, and it removes the problem of our hands. (you asked this in the issue).

csoss_validate, takes the scoring and cv as parameter and also the estimator, and does all relevant checks. So we shouldn't have to.

After we obtain the score, we should save it internally, so that we can determine the performance shifts